packages/devtools_app/lib/src/performance/flutter/performance_screen.dart

```diff
@@ -131,37 +153,86 @@ class _PerformanceScreenBodyState extends State<PerformanceScreenBody>
         ),
       ],
     );
+
+    // We put these two items in a stack because the screen's UI needs to be
+    // built before offline data is processed in order to initialize listeners
+    // that respond to data processing events. The spinner hides the screen's
+    // empty UI while data is being processed.
+    return Stack(
+      children: [
+        performanceScreen,
+        if (loadingOfflineData)
+          Container(
+            color: Theme.of(context).scaffoldBackgroundColor,
+            child: const Center(
+              child: CircularProgressIndicator(),
+            ),
+          ),
+      ],
+    );
   }
 
-  Widget _buildStateControls(PerformanceController controller) {
-    const double includeTextWidth = 600;
+  Widget _buildPerformanceControls() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        _buildPrimaryStateControls(),
+        _buildSecondaryControls(),
+      ],
+    );
+  }
 
+  Widget _buildPrimaryStateControls() {
     return Row(
       children: [
         recordButton(
           key: PerformanceScreen.recordButtonKey,
           recording: recording,
-          includeTextWidth: includeTextWidth,
+          includeTextWidth: _primaryControlsMinIncludeTextWidth,
           onPressed: controller.startRecording,
         ),
         const SizedBox(width: denseSpacing),
         stopRecordingButton(
           key: PerformanceScreen.stopRecordingButtonKey,
           recording: recording,
-          includeTextWidth: includeTextWidth,
+          includeTextWidth: _primaryControlsMinIncludeTextWidth,
           onPressed: controller.stopRecording,
         ),
         const SizedBox(width: defaultSpacing),
         clearButton(
           key: PerformanceScreen.clearButtonKey,
-          includeTextWidth: includeTextWidth,
+          includeTextWidth: _primaryControlsMinIncludeTextWidth,
           busy: recording,
           onPressed: controller.clear,
         ),
       ],
     );
   }
 
+  Widget _buildSecondaryControls() {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      children: [
+        const ProfileGranularityDropdown(PerformanceScreen.id),
+        const SizedBox(width: defaultSpacing),
+        Container(
+          height: Theme.of(context).buttonTheme.height,
+          child: OutlineButton(
+            onPressed: controller.cpuProfileData != null &&
+                    !controller.cpuProfileData.isEmpty
+                ? _exportPerformance
+                : null,
+            child: const MaterialIconLabel(
+              Icons.file_download,
+              'Export',
+              includeTextWidth: _secondaryControlsMinIncludeTextWidth,
+            ),
+          ),
+        ),
```

Comment:
pull this container out into a separate stateless widget class `ExportButton` and put it in common_widgets.dart. Then use an instance of ExportButton in both the performance and timeline screens

Reply:
Should it be named exportButton? Seems like the other stateless widgets are named as camelCase (e.g. stopButton, refreshButton).

Reply:
widget names should not start with a lower case letter. If we have widgets that start with a lower case letter those should be fixed. Likely you are seeing some helper methods that aren't actually widgets.